File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = "aeiouAEIOU"
        i = 0
        j = len(s)-1
        lst = list(s)

        while i < j:
            while vowels.find(lst[i]) == -1 and i < j:
                i+=1
            
            while vowels.find(lst[j]) == -1 and i < j:
                j-=1
            
            lst[i], lst[j] = lst[j], lst[i]
            
            i+=1
            j-=1
        
        return ''.join(lst)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.go
func reverseVowels(s string) string {
    l, r := 0, len(s) - 1 
    ans := []byte(s)
    for l < r {
        for l < r && !isVowel(ans[l]) {
            l += 1
        }

        for l < r && !isVowel(ans[r]) {
            r -= 1
        }

        ans[l], ans[r] = ans[r], ans[l]
        l, r = l + 1, r - 1
    }

    return string(ans)
}

func isVowel(s byte) bool {
    vowels := []byte{'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
    for _, v := range vowels {
        if s == v {
            return true
        }
    }

    return false
}